### PR TITLE
Add Gtk.Window::fullscreened

### PR DESCRIPTION
**Pull request by aleksrutins**
This adds the [fullscreened](https://docs.gtk.org/gtk4/property.Window.fullscreened.html) property to `Window`. It's implemented using `gtk_window_fullscren` and `gtk_window_unfullscreen`.

### CODE REVIEW

**Comment by can-lehmann** [comment]
Looks good so far. Let me know when this is ready for review.

**Comment by aleksrutins** [comment]
After trying it in C++ and getting the same exact bug I was trying to debug (fullscreening momentarily and then returning to normal), I think it's a problem with how I'm using it.

File: owlkettle/widgets.nim [hunk 1]
```diff
@@ -128,17 +128,28 @@ renderable BaseWindow of BaseWidget:
 
 renderable Window of BaseWindow:
   title: string
+  fullscreened: bool ## Whether or not the window is fullscreened; setting this property is equivalent to calling `gtk_window_fullscreen` and `gtk_window_unfullscreen` for values of `true` and `false`, respectively. Either operation is asynchronous, which means you will need to connect to the `::notify` signal in order to know whether the operation was successful (it could fail if, for instance, a window manager refuses to respond to the signal).
```
Comment (can-lehmann) [inline owlkettle/widgets.nim:131]: The docstring here seems to be copied from the GTK documentation. Since GTK itself it LGPL (and the docs are probably too, not sure), we cannot include it in owlkettle. The provided information also does not apply to owlkettle, since it is impossible to connect to the `::notify` signal in owlkettle. I would just remove the comment entirely.

File: owlkettle/widgets.nim [hunk 2]
```diff
@@ -128,17 +128,28 @@ renderable BaseWindow of BaseWidget:
...
   titlebar: Widget ## Custom widget set as the titlebar of the window
   child: Widget
   
   hooks:
     beforeBuild:
       state.internalWidget = gtk_window_new(GTK_WINDOW_TOPLEVEL)
+    afterBuild:
+      if state.fullscreened:
+        gtk_window_fullscreen(state.internalWidget)
```
Comment (can-lehmann) [inline owlkettle/widgets.nim:140]: Why is this required?

Comment (aleksrutins) [reply owlkettle/widgets.nim:140]: That was some confused debugging that I forgot to remove; I'll take it out.

File: owlkettle/widgets.nim (hunk 1 above)
Comment (aleksrutins) [reply owlkettle/widgets.nim:131]: Ah, I didn't realize that. I'll remove it.

**Comment by can-lehmann** [comment]
Looks good!
